webpack/scenes/Subscriptions/Details/SubscriptionDetailEnabledProducts.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router';
+import { notify } from '../../../move_to_foreman/foreman_toast_notifications';
+import { Row, Col, ListView } from 'patternfly-react';
+
+const SubscriptionDetailEnabledProducts = ({ enabledProducts }) => {
+  const listItems = enabledProducts.results.map(product => ({
+    index: product.id,
+    title: product.name,
+    available_content: (
+      product.available_content.map(c => (
+        {
+          enabled: c.enabled,
+          ...c.content,
+        }
+      ))
+    ),
+  }));
+
+  if (listItems.length > 0) {
+    return (
+      <ListView>
+        {listItems.map(
+          (
+            {
+              index,
+              title,
+              available_content,
+              expandedContentText,
+            }
+          ) =>
+            <ListView.Item
+              key={index}
+              heading={title}
+              hideCloseIcon={true}
+            >
+              <Col sm={12}>
+                {available_content.map(content => (
+                  <Row key={content.id}>
+                    <Col sm={12}>
+                      <Row><u>{content.name}</u></Row>
+                    </Col>
+                    <Col sm={3}>
+                      <Row>Content Download URL</Row>
+                      <Row>GPG Key URL</Row>
+                      <Row>Repo Type</Row>
+                      <Row>Enabled?</Row>
+                    </Col>
+                    <Col sm={9}>
+                      <Row>{content.content_url}</Row>
+                      <Row>{content.gpg_url}</Row>
+                      <Row>{content.type}</Row>
+                      <Row>{content.enabled ? __('yes') : __('no')}</Row>
+                    </Col>
+                  </Row>
```

Comment:
You *could* break this out into a component, I'm a fan of having the Row part of a table/list be an individual component for readability. But I don't consider that a blocker, so I will leave that up to you :wink:

Reply:
I agree